go/tools/gazelle/rules/resolve.go

```diff
@@ -56,3 +55,36 @@ func (l label) String() string {
 	}
 	return fmt.Sprintf("%s//%s:%s", repo, l.pkg, l.name)
 }
+
+func NewLabelResolver(c *config.Config) LabelResolver {
+	var e LabelResolver
+	switch c.DepMode {
+	case config.ExternalMode:
+		e = newExternalResolver()
+	case config.VendorMode:
+		e = vendoredResolver{}
+	}
+
+	return &unifiedResolver{
```

Comment:
Was trying to work out how this fits in to this cl, is this the start of an auto mode?